lib/if.c

```diff
@@ -216,6 +216,33 @@ void if_delete(struct interface *ifp)
 	XFREE(MTYPE_IF, ifp);
 }
 
+/* Build the table key */
+static void if_build_key(uint32_t ifindex, struct prefix *p)
```

Comment:
Why two identical static functions? It would be better to reuse the same `if_build_key()` that already exists in the _interface.c_ file.

Also, using a route table to store interfaces is utterly bizarre at best. While you're working on this part of the code, it would be nice if you could rewrite this using a more appropriate data structure like rb-trees.

Reply:
I moved bck to zebra.
so this should not be necessary to move/change if_build_key too